generators/aws-sdk-rbs-generator/lib/aws-sdk-rbs-generator/shape_dictionary/shape.rb

```diff
@@ -122,7 +122,11 @@ def to_rbs_returns
           "bool"
         when "blob"
           if streaming?
-            "::IO"
+            if input?
+              "::String | ::StringIO | ::File"
```

Comment:
while documentation may say "StringIO or File", I believe that sockets would also work ([seems that they only need to answer to #read and #rewind](https://github.com/aws/aws-sdk-ruby/blob/bbe581191d8d0dc9be0f63be5b3656d655c5b8e0/gems/aws-sdk-core/lib/seahorse/client/http/request.rb#L51-L55). Perhaps `::String | ::StringIO | (_Reader & _Rewindable)` would be more accurate, even if more verbose?